packages/core/src/logger/apiLogger.ts

```diff
@@ -0,0 +1,340 @@
+import {
+  HttpRequest,
+  HttpResponse,
+  ApiLoggerInterface,
+  LoggerInterface,
+  LoggingOptions,
+  HttpRequestLoggingOptions,
+  HttpMessageLoggingOptions,
+  LogLevel,
+} from '../coreInterfaces';
+import {
+  CONTENT_LENGTH_HEADER,
+  CONTENT_TYPE_HEADER,
+  getHeader,
+  setHeader,
+} from '../http/httpHeaders';
+import { NullLogger } from './nullLogger';
+
+/**
+ * Represents a logger implementation for API logging.
+ * This logger provides methods to log HTTP requests and responses.
+ */
+export class ApiLogger implements ApiLoggerInterface {
+  private readonly _loggingOptions: LoggingOptions;
+  private readonly _logger: LoggerInterface;
+
+  /**
+   * Constructs a new instance of ApiLogger.
+   * @param loggingOpt The logging options for configuring the logger behavior.
+   */
+  constructor(loggingOpt: LoggingOptions) {
+    this._loggingOptions = loggingOpt;
+    this._logger = loggingOpt.logger ?? new NullLogger();
```

Comment:
`loggingOpt.logger` is never null or undefined according to the `LoggingOptions` type.

Reply:
yes the check here will not be needed anymore. I have removed it.